addons/project/static/src/project_sharing/components/depend_on_ids_one2many/depend_on_ids_one2many_field.js

```diff
@@ -0,0 +1,20 @@
+/** @odoo-module */
+
+import { registry } from "@web/core/registry";
+import { X2ManyField, x2ManyField } from "@web/views/fields/x2many/x2many_field";
+
+import { DependOnIdsListRenderer } from "./depend_on_ids_list_renderer";
+
+export class DependOnIdsOne2ManyField extends X2ManyField {}
+
+DependOnIdsOne2ManyField.components = {
+    ...X2ManyField.components,
+    ListRenderer: DependOnIdsListRenderer,
+};
```

Comment:
```suggestion
export class DependOnIdsOne2ManyField extends X2ManyField {
    static components = {
        ...X2ManyField.components,
        ListRenderer: DependOnIdsListRenderer,
    };
}
```
I think is preferred